server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -980,15 +988,20 @@ public int skippedOperations() {
         }
 
         @Override
-        public Operation next() throws IOException {
+        public Operation next() throws IOException, MissingHistoryOperationsException {
             Translog.Operation op;
             while ((op = delegate.next()) != null) {
                 if (fromSeqNo <= op.seqNo() && op.seqNo() <= toSeqNo) {
+                    opsCount++;
                     return op;
                 } else {
                     filteredOpsCount++;
                 }
             }
+            if(requiredFullRange && (toSeqNo - fromSeqNo +1) != opsCount) {
```

Comment:
Why do you need to do this in the end? Isn't tlog already ordered. You can probably get it immediately?

Reply:
No, Based on the tests, Tlog isn't ordered based on `seqno`